test/include/autoinstall/AutoinstallAskInclude_test.rb

```diff
@@ -0,0 +1,292 @@
+require_relative "../../../src/modules/Profile"
+
+describe "Yast::AutoinstallAskInclude" do
+  module DummyYast
+    class AutoinstallAskClient < Yast::Client
+      def main
+        Yast.include self, "autoinstall/ask.rb"
+      end
+
+      def initialize
+        main
+      end
+    end
+  end
+
+  let(:client) { DummyYast::AutoinstallAskClient.new }
+  let(:base_ask) do
+    { "type" => "string", "question" => "hostname?", "default" => "my.site.de",
+      "help" => "Some help", "dialog" => 0, "element" => 0,
+      "stage" => "initial" }
+  end
+  let(:profile) { { "general" => { "ask-list" => ask_list } } }
+  let(:result) { :ok }
+
+  describe "#askDialog" do
+    let(:ask_list) { [ask] }
+
+    before(:each) do
+      allow(Yast::Profile).to receive(:current).and_return(profile)
+      allow(Yast::Stage).to receive(:initial).and_return(true)
+      allow(Yast::UI).to receive(:UserInput).and_return(result)
+    end
+
+    context "when ask-list is empty" do
+      let(:ask_list) { [] }
+
+      it "no dialog is shown" do
+        expect(Yast::UI).to_not receive(:OpenDialog)
+        client.askDialog
+      end
+    end
+
+    describe "dialogs creation" do
+      context "when the ask-list contains a question with type 'string'" do
+        let(:ask) { base_ask }
+
+        it "creates a TextEntry widget" do
+          expect(Yast::UI).to receive(:OpenDialog).and_call_original
+          expect(client).to receive(:TextEntry).
+            with(client.Id("0_0"), client.Opt(:notify), ask["question"], ask["default"]).
+            and_call_original
+          client.askDialog
+        end
+      end
+
+      context "when ask-list contains a question with type 'selection'" do
+        let(:ask) { base_ask.merge("selection" => items, "default" => "desktop") }
+        let(:items) {
+          %w(desktop server).map { |i| { "value" => i, "label" => i.capitalize } }
+        }
+
+        it "creates a ComboBox widget" do
+          expect(Yast::UI).to receive(:OpenDialog).and_call_original
+          expected_options = [
+            client.Item(client.Id("desktop"), "Desktop", true),
+            client.Item(client.Id("server"), "Server", false)
+          ]
+          expect(client).to receive(:ComboBox).
+            with(client.Id("0_0"), client.Opt(:notify), ask["question"], expected_options).
+            and_call_original
+          client.askDialog
+        end
+      end
+
+      context "when ask-list contains a question with type 'password'" do
+        let(:ask) { base_ask.merge("password" => true) }
+
+        it "creates two Password widgets" do
+          expect(Yast::UI).to receive(:OpenDialog).and_call_original
+          expect(client).to receive(:Password).
+            with(client.Id("0_0"), client.Opt(:notify), ask["question"], ask["default"]).
+            and_call_original
+          expect(client).to receive(:Password).
+            with(client.Id(:pass2), client.Opt(:notify), "", ask["default"]).
```

Comment:
So it actually do not work? I do not know autoyast code so good, so I mainly ask.